### PR TITLE
[Admin] Add `orders/index` component with order list table

### PR DESCRIPTION
**Pull request by rainerdema**
## Summary

This PR introduces a new `orders/index` component that is structured similarly to the existing `products/index` component. 

- **Order List:**
  - Displays a list of orders with attributes such as order number, email, total amount, and shipping and billing addresses.
  -  Enable a search across order number, shipment number, billing address name, email, and promotion code value.
  - Add `Only show complete orders` filter
   

https://github.com/solidusio/solidus/assets/19948291/a1890a2e-c948-4989-b61e-eeed532d4bd4

Screenshot 2023-08-23 at 19 07 50

Screenshot 2023-08-23 at 19 07 57

Screenshot 2023-08-24 at 15 31 18

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: admin/app/components/solidus_admin/orders/index/component.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        <%= t(".feedback_description") %>
+      </span>
+      <%= render @button_component.new(
+        tag: :a,
+        text: t(".give_feedback"),
+        href: "https://solidus.io/feedback?category=solidus-admin",
+        icon: "feedback-line",
+        scheme: :secondary,
+        target: :_blank,
+      ) %>
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.html.erb:18]: Given that we'll need this button for every page, what do you think about extracting it into a component?

File: admin/app/components/solidus_admin/orders/index/component.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+class SolidusAdmin::Orders::Index::Component < SolidusAdmin::BaseComponent
+  def initialize(
+    page:,
+    badge_component: component('ui/badge'),
+    table_component: component('ui/table'),
+    pagination_component: component('ui/table/pagination'),
+    button_component: component("ui/button")
+  )
+    @page = page
+
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.rb:12]: Super-nit: we can remove this blank line :slightly_smiling_face:

File: admin/app/components/solidus_admin/orders/index/component.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Spree::Order.model_name.human.pluralize
+  end
+
+  def prev_page_link
+    @page.first? ? nil : solidus_admin.url_for(host: request.host, port: request.port, **request.params, page: @page.number - 1)
+  end
+
+  def next_page_link
+    @page.last? ? nil : solidus_admin.url_for(host: request.host, port: request.port, **request.params, page: @page.next_param)
+  end
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.rb:29]: Not sure if we already attempted that, but do you think that could be extracted to the page or table component, or somewhere else? Or do you think it's better to wait for something like that?

File: admin/app/components/solidus_admin/orders/index/component.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      date_column,
+      customer_column,
+      total_column,
+      items_column,
+      payment_column,
+      shipment_column,
+    ]
+  end
+
+  def number_column
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.rb:57]: Looking at the design files, it seems that the order number needs to be bolded unless the order is paid.

File: admin/app/components/solidus_admin/orders/index/component.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Add your component translations here.
+# Use the translation in the example in your template with `t(".hello")`.
+en:
+  create_order: 'Create Order'
+  give_feedback: 'Give feedback'
+  feedback_description: 'We are constantly trying to improve. Please let us know what you think about this admin page.'
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.yml:6]: Same idea about extracting it into a feedback component.

File: admin/app/components/solidus_admin/orders/index/component.rb [hunk 6]
```diff
@@ -33,7 +33,13 @@ def batch_actions
   end
 
   def filters
-    []
+    [
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.rb:36]: Nit: there's a typo in the corresponding commit message: "between the old and new ~~product~~ order index page".

**Review by waiting-for-dev — COMMENTED** [review]
Thanks, @rainerdema! Great job! I only left some comments.

File: admin/app/components/solidus_admin/orders/index/component.rb (hunk 2 above)
Comment (rainerdema) [reply admin/app/components/solidus_admin/orders/index/component.rb]: The blank line was introduced to help visually separate the initialization of injectable components from other elements 😄
However, I noticed that was already removed in `products/index`.
In the interest of consistency, I'll remove it here too 👍

File: admin/app/components/solidus_admin/orders/index/component.html.erb (hunk 1 above)
Comment (rainerdema) [reply admin/app/components/solidus_admin/orders/index/component.html.erb]: Thanks for the suggestion! I've refactored it into a `ui/feedback` component. However, I'm on the fence about classifying it strictly as a UI component, so let me know what you think 👍

File: admin/app/components/solidus_admin/orders/index/component.yml (hunk 5 above)
Comment (rainerdema) [reply admin/app/components/solidus_admin/orders/index/component.yml]: Done 👍 https://github.com/solidusio/solidus/pull/5347#discussion_r1303073213

File: admin/app/components/solidus_admin/orders/index/component.rb (hunk 3 above)
Comment (rainerdema) [reply admin/app/components/solidus_admin/orders/index/component.rb:32]: We previously tried to move this functionality into the table component, but we found it challenging to manage. As a result, passing the links directly to the component seemed simpler.
I'd suggest keeping it here for now, at least until we finish with the features introductions on these index pages.
Just to clarify, when you mentioned the 'page component', what exactly did you have in mind?

File: admin/app/components/solidus_admin/orders/index/component.rb (hunk 4 above)
Comment (rainerdema) [reply admin/app/components/solidus_admin/orders/index/component.rb:60]: Thank you for catching this! 🙏 
After revisiting the design files, I realized that:
- The order number should indeed be bolded if the order hasn't been paid or shipped.
- When the order is paid and shipped, the row will have a "fading effect" like the header: I've made some adjustments with the introduction of a `fade_row_proc` to ensure this behavior.

Let me know WDYT 👍

File: admin/app/components/solidus_admin/orders/index/component.html.erb (hunk 1 above)
Comment (waiting-for-dev) [reply admin/app/components/solidus_admin/orders/index/component.html.erb]: Yeah, I don't see it as a UI thing. Probably it should be a top-level component, just like the skip link. Maybe at some point we can consider grouping non-UI components in a certain way?

File: admin/app/components/solidus_admin/orders/index/component.rb [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  end
+
+  def number_column
+    {
+      header: :order,
+      data: ->(order) do
+        order_path = spree.edit_admin_order_path(order)
+
+        if !fade_row_proc.call(order)
+          content_tag(:a, order.number, href: order_path, class: 'font-semibold')
```
Comment (waiting-for-dev) [inline admin/app/components/solidus_admin/orders/index/component.rb:67]: Nit: for consistency, why not using the `link_to` helper also in this case?

**Review by waiting-for-dev — APPROVED** [review]
Cool, @rainerdema! Left a couple of non-blocking comments. Leaving to you the call about the feedback component.

**Pull request merged**